Generator/Sources/NeedleFramework/Parsing/Tasks/DeclarationsParserTask.swift

```diff
@@ -79,4 +84,9 @@ private extension Structure {
     var isDependencyProtocol: Bool {
         return inheritedTypes.contains("Dependency") || inheritedTypes.contains("\(needleModuleName).Dependency")
     }
+
+    /// Check if this structure represents the root of a dependency graph.
+    var isRoot: Bool {
+        return inheritedTypeNames.contains("RootComponent") || inheritedTypeNames.contains("\(needleModuleName).RootComponent")
```

Comment:
Would be better performance to use `MyRootComponent: Component<RootDependency>` instead so that we would not need this extra 2 contains calls. However, since we don't know how much this affects overall time, this is fine.